frame/contracts/src/exec.rs

```diff
@@ -1024,6 +1034,14 @@ where
 		self.frames().nth(1).map(|f| &f.account_id).unwrap_or(&self.origin)
 	}
 
+	fn is_contract(&self, address: T::AccountId) -> bool {
+		ContractInfoOf::<T>::get(&address).is_some()
```

Comment:
This will avoid loading the whole value only to check for existence.

```suggestion
		ContractInfoOf::<T>::contains_key(&address)
```